client/src/main/java/io/pravega/client/segment/impl/SegmentOutputStreamImpl.java

```diff
@@ -422,6 +423,24 @@ public void authTokenCheckFailed(WireCommands.AuthTokenCheckFailed authTokenChec
         }
     }
 
+    public SegmentOutputStreamImpl(String segmentName, boolean useConnectionPooling, Controller controller,
+                            ConnectionFactory connectionFactory, UUID writerId, RetryWithBackoff retrySchedule,
+                            String delegationToken) {
+        this.segmentName = segmentName;
+        this.useConnectionPooling = useConnectionPooling;
+        this.controller = controller;
+        this.connectionFactory = connectionFactory;
+        this.writerId = writerId;
+        this.resendToSuccessorsCallback = segment -> {
+            List<PendingEvent> r = this.getUnackedEventsOnSeal();
```

Comment:
This can be done externally to the class. So I think it would be better to put this in the revision stream client, than have two constructors that end up creating classes that behave differently. If it needs to be here for some reason, then it should at the very least should be spelled out in the Javadoc.

Reply:
This change ensures that the default behavior of `SegmentOutputStreamImpl` is to change the state of writer to `Sealed` and ensure all the write futures are completed exceptionally. This change ensures all the users of ` SegmentOutputStream createOutputStreamForSegment(Segment segment, EventWriterConfig config, String delegationToken) ` have the same behavior.   (this is used by RevisionedStreamClient and ByteStreamClient).  Let me know your thoughts on this.

I have updated the java doc for the constructor documenting the behaviour of the default callback.

Reply:
Thanks for the feedback, I have refactored the code and removed the extra constructor.